Source/cmake/OptionsWPE.cmake

```diff
@@ -357,7 +357,7 @@ SET_AND_EXPOSE_TO_BUILD(USE_NICOSIA TRUE)
 SET_AND_EXPOSE_TO_BUILD(USE_ANGLE ${USE_ANGLE_WEBGL})
 SET_AND_EXPOSE_TO_BUILD(HAVE_OS_DARK_MODE_SUPPORT 1)
 
-SET_AND_EXPOSE_TO_BUILD(USE_FULLHD_VIDEO_DECODING_LIMIT ${ENABLE_FULLHD_VIDEO_DECODING_LIMIT})
+add_definitions(-DVIDEO_DECODING_LIMIT="${VIDEO_DECODING_LIMIT}")
```

Comment:
I'm going to suggest changing this from a build option to a runtime API so that you don't need a new build to change the limit.

That said, if you do prefer to stick with a build option, I would define it only for the one file that actually needs it, MediaPlayerPrivateGStreamerMSE.cpp, rather than here. What you have currently is going to require a full rebuild of all of WebKit each time you want to change the limit.

Reply:
IMHO, for the typical use case where we need this feature (to adapt WPE to the capabilities of different devices, typically set-top-boxes) I think there's no advantage on being able to change the limit at runtime and a build option should be enough.

What would be the way to define it for a single file in cmake, as you suggest?

Reply:
You could try this in Source/WebCore/platform/GStreamer.cmake:

```
set_source_files_properties(MediaPlayerPrivateGStreamerMSE.cpp PROPERTIES COMPILE_DEFINITIONS "-DVIDEO_DECODING_LIMIT=${VIDEO_DECODING_LIMIT}")
```

(Note this will work only because it is not unified: you won't be able to add it to unified build in the future.)

> IMHO, for the typical use case where we need this feature (to adapt WPE to the capabilities of different devices, typically set-top-boxes) I think there's no advantage on being able to change the limit at runtime and a build option should be enough.

OK then, though it means you won't be able to enable the video decoding limit in any standard builds, e.g. WebKit flatpak runtime builds, or Linux distro builds. Perhaps a bit disappointing if you were hoping to use the WebKit flatpak runtime on Raspberry Pi, for example?